aspnetcore/fundamentals/minimal-apis/openapi.md

```diff
@@ -77,52 +77,143 @@ ASP.NET Core provides the [`Microsoft.AspNetCore.OpenApi`](https://www.nuget.org
 
-### `WithOpenApi` call on endpoints adds an OpenAPI annotation
+### Add OpenAPI annotations to endpoints via `WithOpenApi`
 
-Calling [`WithOpenApi`](https://github.com/dotnet/aspnetcore/blob/8a4b4deb09c04134f22f8d39aae21d212282004f/src/OpenApi/src/OpenApiRouteHandlerBuilderExtensions.cs) on an endpoint without parameters adds an [OpenAPI annotation](https://github.com/dotnet/aspnetcore/blob/8a4b4deb09c04134f22f8d39aae21d212282004f/src/OpenApi/src/OpenApiRouteHandlerBuilderExtensions.cs#L65-L82) to the endpoints metadata. This metadata can be:
+Calling [WithOpenApi](/dotnet/api/microsoft.aspnetcore.builder.openapiendpointconventionbuilderextensions.withopenapi) on the endpoint to the endpoints metadata. This metadata can be:
 
 * Consumed in third-party packages like [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore/).
 * Displayed in the Swagger user interface or in YAML or JSON generated to define the file.
 
 [!code-csharp[](7.0-samples/todo/Program.cs?name=snippet_withopenapi&highlight=9)]
 
-#### Call `WithOpenApi` with parameters
+#### Modify the OpenAPI annotation in `WithOpenApi`
 
-The [`WithOpenApi`](https://github.com/dotnet/aspnetcore/blob/8a4b4deb09c04134f22f8d39aae21d212282004f/src/OpenApi/src/OpenApiRouteHandlerBuilderExtensions.cs#L49) method accepts a function that can be used to modify the OpenAPI annotation. For example, in the following code, a description is added to the first parameter of the endpoint:
+The [WithOpenApi](/dotnet/api/microsoft.aspnetcore.builder.openapiendpointconventionbuilderextensions.withopenapi) method accepts a function that can be used to directly modify the OpenAPI annotation. For example, in the following code, a description is added to the first parameter of the endpoint:
 
 [!code-csharp[](7.0-samples/todo/Program.cs?name=snippet_withopenapi2&highlight=9-99)]
 
+### Add operation IDs to Open API
+
+Operation IDs are used to uniquely identify a given endpoint in OpenAPI. The [`WithName`](/dotnet/api/microsoft.aspnetcore.builder.routingendpointconventionbuilderextensions.withname) extension method can be used to set the operation ID used for a method.
+
+[!code-csharp[](samples/todo/Program.cs?name=snippet_name)]
+
+Alternatively, the `OperationId` property can be set directly on the OpenApi annotation.
+
+```csharp
+app.MapGet("/todos", async (TodoDb db) => await db.Todos.ToListAsync())
+    .WithOpenApi(operation => new(operation)
+    {
+        OperationId = "GetTodos"
+    });
+```
+
+### Add tags to the Open API description
+
+OpenAPI supports using [tag objects](https://swagger.io/docs/specification/grouping-operations-with-tags/) to categorize operations. These tags are typically used to group operations in the Swagger UI. These tags can be added to an operation by invoking the [WithTags](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.withtags) extension method on the endpoint with the desired tags.
```

Comment:
Should we link to the OpenAPI spec instead of swagger.io perhaps?

Reply:
The Swagger docs are a little bit more helpful at informing the user _how_ tags should be used instead of just telling users what they are so they feel more helpful to link out to.

Reply:
OK. I admit something about linking to a commercial site (that isn't ms.com) turns me off.